engines/sci/graphics/palette32.cpp

```diff
@@ -21,20 +21,156 @@
  */
 
 #include "common/file.h"
+#include "common/system.h"
+#include "graphics/palette.h"
 
 #include "sci/sci.h"
+#include "sci/resource.h"
 #include "sci/graphics/palette32.h"
+#include "sci/graphics/screen.h"
 
 namespace Sci {
 	
 GfxPalette32::GfxPalette32(ResourceManager *resMan, GfxScreen *screen)
-	: GfxPalette(resMan, screen), _clutTable(0), _cyclers(), _cycleMap() {}
+	: GfxPalette(resMan, screen),
+	_clutTable(nullptr),
+	// Palette cycling
+	_cyclers(), _cycleMap(),
+	// Palette varying
+	_varyTime(0), _varyDirection(0), _varyTargetPercent(0),
+	_varyTargetPalette(nullptr), _varyStartPalette(nullptr),
+	_varyFromColor(0), _varyToColor(255), _varyNumTimesPaused(0),
+	_varyPercent(_varyTargetPercent), _varyLastTick(0),
+	// Palette versioning
+	_version(1), _needsUpdate(false) {
+		// TODO: Sci::Palette should have a default constructor that does
+		// all of this
+		for (int color = 0; color < ARRAYSIZE(_sysPalette.colors); color++) {
+			_sourcePalette.colors[color] = _nextPalette.colors[color] = _sysPalette.colors[color];
+			// TODO: Intensity and mapping are not used by SCI32 but are used by
+			// other parts of the graphic engine right now. Replace those parts.
+			_sourcePalette.intensity[color] = _nextPalette.intensity[color] = 100;
+			_sourcePalette.mapping[color] = _nextPalette.mapping[color] = color;
+		}
+
+		for (int i = 0; i < ARRAYSIZE(_fadeTable); ++i) {
+			_fadeTable[i] = 100;
+		}
+
+		// TODO: In SCI engine, GfxPalette (PaletteMgr) constructor handles
+		// loading the default palette, but ScummVM already loads it from
+		// somewhere else, so should ScummVM be updated to match SCI or should
+		// it be left the way it is already?
+		assert(kernelSetFromResource(999, true));
```

Comment:
Well, we initialize the default palette in SciEngine::initGraphics() via setDefault(), which occurs during engine initialization. We can just do that in the constructor. 

This was done in commit dc8b826f5ea77a0173c5ad64d6508f9ec7ed798a, in order to fix the palette during MT-32 emulation, where some MT-32 related messages can be set when starting a game. So, we can just leave it the way it is already, as the end result is the same,